### PR TITLE
backend: Refactor model deployments folder

### PR DESCRIPTION
**Pull request by tianjing-li**
- Refactors the `model_deployments` folder to move it into the parent `backend` folder
- Changes `models` to `database_models`

### CODE REVIEW

File: src/community/model_deployments/hugging_face.py [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import logging
-from typing import Any, Dict, Generator, List
+from typing import Any, Dict, List
 
-from cohere.types import NonStreamedChatResponse, StreamedChatResponse
 from transformers import AutoModelForCausalLM, AutoTokenizer
 
+from backend.model_deployments.base import BaseDeployment
```
Comment (lusmoura) [inline src/community/model_deployments/hugging_face.py:6]: Should we import this from community as it was before?

File: src/community/model_deployments/local_model.py [hunk 2]
```diff
@@ -1,10 +1,9 @@
-import logging
 from typing import Any, Dict, List
 
 from llama_cpp import Llama
 
+from backend.model_deployments.base import BaseDeployment
```
Comment (lusmoura) [inline src/community/model_deployments/local_model.py:5]: Same as above

**Review by lusmoura — APPROVED** [review]

File: src/community/model_deployments/hugging_face.py (hunk 1 above)
Comment (tianjing-li) [reply src/community/model_deployments/hugging_face.py]: done

File: src/community/model_deployments/local_model.py (hunk 2 above)
Comment (tianjing-li) [reply src/community/model_deployments/local_model.py]: done

**Pull request merged**